ext/nnstreamer/tensor_filter/tensor_filter_caffe2_core.cc

```diff
@@ -0,0 +1,465 @@
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::setOutputTensorProp (const GstTensorFilterProperties * prop)
+{
+  gst_tensors_info_copy (&outputTensorMeta, &prop->output_meta);
+  return 0;
+}
+
+/**
+ * @brief	return the Dimension of Input Tensor.
+ * @param[out] info Structure for tensor info.
+ * @todo return whole array rather than index 0
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::getInputTensorDim (GstTensorsInfo * info)
+{
+  gst_tensors_info_copy (info, &inputTensorMeta);
+  return 0;
+}
+
+/**
+ * @brief	return the Dimension of Tensor.
+ * @param[out] info Structure for tensor info.
+ * @todo return whole array rather than index 0
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::getOutputTensorDim (GstTensorsInfo * info)
+{
+  gst_tensors_info_copy (info, &outputTensorMeta);
+  return 0;
+}
+
+/**
+ * @brief	run the model with the input.
+ * @param[in] input : The array of input tensors
+ * @param[out]  output : The array of output tensors
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::run (const GstTensorMemory * input, GstTensorMemory * output)
+{
+  int i;
+#if (DBG)
+  gint64 start_time = g_get_real_time ();
+#endif
+  for (i = 0; i < inputTensorMeta.num_tensors; i++){
+    Tensor *inputTensor = inputTensorMap.
+                            find(inputTensorMeta.info[i].name)->second;
+    inputTensor->ShareExternalPointer ((float*) input[i].data);
+    switch (outputTensorMeta.info[i].type){
+      case _NNS_INT32:
+        inputTensor->ShareExternalPointer ((int32_t*) input[i].data);
+        break;
+      case _NNS_UINT32:
+        g_critical ("invalid data type is used");
+        return -1;
+      case _NNS_INT16:
+        inputTensor->ShareExternalPointer ((int16_t*) input[i].data);
+        break;
+      case _NNS_UINT16:
+        inputTensor->ShareExternalPointer ((uint16_t*) input[i].data);
+        break;
+      case _NNS_INT8:
+        inputTensor->ShareExternalPointer ((int8_t*) input[i].data);
+        break;
+      case _NNS_UINT8:
+        inputTensor->ShareExternalPointer ((uint8_t*) input[i].data);
+        break;
+      case _NNS_FLOAT64:
+        inputTensor->ShareExternalPointer ((double*) input[i].data);
+        break;
+      case _NNS_FLOAT32:
+        inputTensor->ShareExternalPointer ((float*) input[i].data);
+        break;
+      case _NNS_INT64:
+        inputTensor->ShareExternalPointer ((int64_t*) input[i].data);
+        break;
+      case _NNS_UINT64:
+        g_critical ("invalid data type is used");
+        return -1;
+      default:
+        g_critical ("invalid data type is used");
+        return -1;
+    }
+  }
+
+  workSpace.RunNet (predictNet.name ());
+
+  for (i = 0; i < outputTensorMeta.num_tensors; i++) {
+    const auto& out = workSpace.GetBlob (outputTensorMeta.info[i].name)
+      ->Get<Tensor> ();
+    switch (outputTensorMeta.info[i].type){
+      case _NNS_INT32:
+        output[i].data = out.data<int32_t>();
```

Comment:
If my understanding is correct, as `allocate_in_invoke` is set to `FALSE`, `GstTensorMemory * output` already has memory allocated, overwriting `output[i].data` will result in loss of that memory.
I think `allocate_in_invoke` should be set to `TRUE`

Reply:
it's changed.